p2panda-sync/src/cbor.rs

```diff
@@ -56,20 +54,23 @@ where
     type Item = T;
     type Error = SyncError;
 
-    fn decode(
-        &mut self,
-        src: &mut tokio_util::bytes::BytesMut,
-    ) -> Result<Option<Self::Item>, Self::Error> {
+    fn decode(&mut self, src: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
         let reader = src.reader();
-        let result: Result<Self::Item, _> = ciborium::from_reader(reader);
+        let result: Result<Self::Item, _> = decode_cbor(reader);
         match result {
             // If we read the item, we also need to advance the underlying buffer.
             Ok(item) => Ok(Some(item)),
-            Err(ref error) => match error {
-                // Sometimes the EOF is signalled as IO error
-                ciborium::de::Error::Io(_) => Ok(None),
-                e => Err(SyncError::Codec(e.to_string())),
-            },
+            Err(ref error) => {
+                match error {
+                    DecodeError::Io(err) => {
+                        // @TODO
+                        panic!("{}", err);
+                    }
+                    err => Err(SyncError::RemoteUnexpectedBehaviour(format!(
+                        "CBOR codec failed decoding message from remote peer, {err}"
+                    ))),
```

Comment:
If we failed decoding a remote peer's messages then we can consider this a protocol violation